### ISSUE TITLE
custom dataset detection accuracy

### ISSUE THREAD

**Issue by DindinGS** [AlexeyAB/darknet#3128]
I am training a custom dataset that has only two classes. However, both two classes have 5 different stages. The original dataset was imbalance (2 stages have more entries than the other 2) so I tried augmenting them so they have similar number. The resulting dataset is around 400 images per stage.  Around 5300 training images overall including 800 negative images per class. 

Augmentation: rotation, magnification (80-120%) only
Split: 70-30 train-test after separating them into stages (Then augmetation)
One of the stages do actually have multiple boxes in a single image - around 5-8, small but I ensured to be at least 32px x 32px bounding box- 
while the others only have 1 - around 5 to 10x larger than small.
One stage I considered as mix (stage w/ multiple + stage w/ one instance).

I'm currently training and it's around 4k iterations. I'm planning to run it for 10k iters (64 subd, 32  batch (orig at 16 until 4k), 416x416, 1 gpu, LR 0.001 to change 0.0001 at 8k iter / 100 epochs). 
My last mAP check at 4300 is
 for thresh = 0.25, precision = 0.80, recall = 0.86, F1-score = 0.83
 for thresh = 0.25, TP = 2676, FP = 671, FN = 426, average IoU = 57.90 %

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall
 mean average precision (mAP@0.50) = 0.842107, or 84.21 %

For my last few iterations (5k) before I restarted to increase the learning rate, my mAP is getting stuck at 83-86%. 
(Previous training: When I only augmented them equally, around 20 augments per original image, I can only get a maximum of 80% mAP - 8k iterations)

I tried recalculating anchors on my previous training but  I don't see any improvements. Also tried using spp but it just made the training slower (also w/ same accuracy). I say that the 416x416 dim is better than 608x608 (faster training, really similar accuracy) but maybe I'll try it again in 8k iters. I'm only training with a GPU server so I can't show any mAP-loss graph. Current loss is oscillating at 0.1-0.5.

**Questions (I'm trying to at least reach 90+% FN):**
- Should I augment each stage with 2k images even if I only have two classes? (original dataset is actually at 150 images per class)
- Do I need to increase the negatives? This is a medical-related classification so I'm trying to focus on the FN but I don't want to sacrifice the FP that much.
- Will this 84-86% still improve with just continuous training? I have been running this task for over five months now and previous trainings only reached 20k iters with similar mAPs at 7k iters.
- I'm using focal_loss = 1 but I haven't tried removing it to check if it helps. Should I remove it?

UPDATE AT 4700 iter (Precision is getting left behind~~):
 for thresh = 0.25, precision = 0.80, recall = 0.88, F1-score = 0.84
 for thresh = 0.25, TP = 2719, FP = 662, FN = 383, average IoU = 58.16 %

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall
 mean average precision (mAP@0.50) = 0.844686, or 84.47 %
